### PR TITLE
Add performance monitoring API

### PR DESCRIPTION
**Pull request by tustanivsky**
Related to #13

### CODE REVIEW

File: plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+void SentrySpanAndroid::SetTag(const FString& key, const FString& value)
+{
+	CallMethod<void>(SetTagMethod, *GetJString(key), *GetJString(value));
+}
+
+void SentrySpanAndroid::RemoveTag(const FString& key)
+{
+	SetTag(key, TEXT(""));
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp:39]: TIL.
Do you happen to know where this gets filtered? Is that tag still part of the event or does it get removed by the SDK?

File: plugin-dev/Source/Sentry/Private/Apple/SentrySubsystemApple.cpp [hunk 2]
```diff
@@ -65,6 +68,15 @@ void SentrySubsystemApple::InitWithSettings(const USentrySettings* settings, USe
...
 			[options addInAppExclude:it->GetNSString()];
 		}
+		options.enableTracing = settings->EnableTracing;
+		if(settings->EnableTracing && settings->SamplingType == ESentryTracesSamplingType::UniformSampleRate)
+		{
+			options.tracesSampleRate = [NSNumber numberWithFloat:settings->TracesSampleRate];
+		}
+		if(settings->EnableTracing && settings->SamplingType == ESentryTracesSamplingType::TracesSampler)
+		{
+			options.tracesSampler = nil;
```
[the remark this answers is not in the document]
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Apple/SentrySubsystemApple.cpp:78]: Currently `TracesSampler` feature is not supported by the UE plugin so this is just a stub here

File: plugin-dev/Source/Sentry/Private/Apple/SentryTransactionApple.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+bool SentryTransactionApple::IsFinished() const
+{
+	return TransactionApple.isFinished;
+}
+
+void SentryTransactionApple::SetName(const FString& name)
+{
+	// no corresponding implementation in sentry-cocoa
```
[the remark this answers is not in the document]
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Apple/SentryTransactionApple.cpp:43]: Yes, that's a good point. I'll add a corresponding log message

File: plugin-dev/Source/Sentry/Private/Apple/SentrySubsystemApple.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bitsandfoxes) [reply plugin-dev/Source/Sentry/Private/Apple/SentrySubsystemApple.cpp:78]: Can we log this here like it's done for desktop?

File: plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+void SentrySpanAndroid::Finish()
+{
+	CallMethod<void>(FinishMethod);
+}
+
+bool SentrySpanAndroid::IsFinished() const
+{
+	return false;
```
[the remark this answers is not in the document]
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp]: Somehow I've overlooked this, good catch!

File: plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp (hunk 1 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp:39]: Based on a previous experience with setting extra context for captured events I assume that the "removed" tag is still sent to Sentry however it's not displayed due to its empty value. I wasn't able to find any code in sentry-java that removes it explicitly on the client side but it's hardly confirms my theory

**Review by bitsandfoxes — APPROVED** [review]
:shipit:

**Pull request merged**